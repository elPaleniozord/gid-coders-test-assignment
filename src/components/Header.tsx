const Header = () => {
  return (
    <header>
      <h1>BREWDOG BREWERY</h1>
      <p>Explore BREWDOG beer catalogue!</p>
    </header>
  )
}

export default Header