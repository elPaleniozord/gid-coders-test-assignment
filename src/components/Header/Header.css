.header-container {
  background-color: var(--highlight);
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  align-items: center;
  z-index: 1;
  transition: all .5s ease-in-out;
  transition-delay: .4s;
  position: fixed;
  top: 0;
  left: 0;
  height: 50px;
  width: 100vw;
  opacity: 0;
}
.show {
  opacity: 1;
}

.header-caption {
  font-size: 2.6rem;
  margin: 0 .3rem 0 .3rem;
}

.header-logo {
  height: 2.6rem;
}