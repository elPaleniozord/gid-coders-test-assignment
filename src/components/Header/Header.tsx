import useScrollPosition from '../../hooks/useScrollPosition'
import './Header.css'

const Header = () => {
  const visible = useScrollPosition()
  console.log(visible)
  return (
    <header className={`header-container ${visible && 'header-container__big'}`}>
      <h1 className="header-caption">BREWDOG</h1>
      <img className="header-logo" src={`${process.env.PUBLIC_URL}/brewdog-logo.png`} alt="brewdog-logo"></img>
      <p className="header-description">Explore BREWDOG brewery catalogue!</p>
    </header>
  )
}

export default Header