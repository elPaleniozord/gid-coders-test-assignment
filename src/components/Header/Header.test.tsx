import { render } from '@testing-library/react';
import Header from "./Header"

test('should render Header Component correctly', () => {
  const { container } = render(<Header scroll={0}/>)
  const header = container.querySelector('header')

  expect(header).toMatchSnapshot()
}) 