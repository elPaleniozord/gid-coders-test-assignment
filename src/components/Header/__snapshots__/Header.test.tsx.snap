// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header Component correctly 1`] = `
<header
  class="header-container "
>
  <h1
    class="header-caption"
  >
    BREWDOG
  </h1>
  <img
    alt="brewdog-logo"
    class="header-logo"
    src="/brewdog-logo-small.png"
  />
</header>
`;
